gradle/testDependencies.gradle

```diff
@@ -36,4 +36,8 @@ testLibraries.mockito_kotlin = 'com.nhaarman:mockito-kotlin:1.6.0'
 testLibraries.mockito_kotlin2 = 'com.nhaarman.mockitokotlin2:mockito-kotlin:2.1.0'
 testLibraries.jackson_kotlin = 'com.fasterxml.jackson.module:jackson-module-kotlin:2.9.2'
 testLibraries.archunit = "com.tngtech.archunit:archunit:0.11.0"
-testLibraries.archunit_junit4 = "com.tngtech.archunit:archunit-junit4:0.11.0"
\ No newline at end of file
+testLibraries.archunit_junit4 = "com.tngtech.archunit:archunit-junit4:0.11.0"
+testLibraries.guava_testlib = 'com.google.guava:guava-testlib:' + libraries.guava.version
+testLibraries.quickcheck_core = 'com.pholser:junit-quickcheck-core:0.9'
+testLibraries.quickcheck_generators = 'com.pholser:junit-quickcheck-generators:0.9'
+testLibraries.bricks_automation = 'dk.brics:automaton:1.12-1'
```

Comment:
The tests are incredibly detailed which is really awesome. However, I'm hesitant by the new libraries they require. Could the same kind of coverage be achieved differently? //cc @big-guy

Reply:
@lacasseio , I support your view, however, my answer to the same question was

1) `guava:guava-testlib` is the only available library that **automatically** tests `Map` implementations. I'm afraid it can't really be replaced without writing **lots** of tests manually.

2) `quickcheck` can be replaced with hand-baked `random-driven` test, however, `quickcheck` is known at StackOverflow, and `quickcheck` implements automatic shrinking of the reproducers. That helps a lot for debugging of the failures => it simplifies the maintenance.

3) `dk.brics:automaton` is a "well known" automata-driven regexp engine. It enables to write test cases like "I want a sting that conforms to this regexp". I gues that simplifies the understanding of the tests. Of course I can create hand-rolled implementation for the specific case in question, however I think automation-based approach is simpler.

Note: I would like to incorporate bricks_automation-driven regexp generator right into quickcheck, however it is not clear what the release schedule for quickcheck is, so I just wen ahead and added one (or two) classes right into the test code.

Reply:
At the end of the day, `testLibraries` do not sneak to the resulting binary.